sdk/communication/AzureCommunicationChat/Source/Models/Message.swift

```diff
@@ -60,17 +60,29 @@ public struct Message: Codable {
     ///   - chatMessage: The ChatMessage to initialize from.
     public init(
         from chatMessage: ChatMessage
-    ) {
+    ) throws {
         self.id = chatMessage.id
         self.type = chatMessage.type
         self.sequenceId = chatMessage.sequenceId
         self.version = chatMessage.version
-        self.content = (chatMessage.content != nil) ? MessageContent(from: chatMessage.content!) : nil
+
+        // Convert ChatMessageContent to MessageContent
+        if let content = chatMessage.content {
+            self.content = try MessageContent(from: content)
+        } else {
+            self.content = nil
+        }
+
         self.senderDisplayName = chatMessage.senderDisplayName
         self.createdOn = chatMessage.createdOn
-        self
-            .sender = (chatMessage.senderId != nil) ? CommunicationUserIdentifier(identifier: chatMessage.senderId!) :
-            nil
+
+        // Deserialize the identifier model to CommunicationIdentifier
+        if let identifierModel = chatMessage.senderCommunicationIdentifier {
+            self.sender = try IdentifierSerializer.deserialize(identifier: identifierModel)
+        } else {
+            self.sender = nil
+        }
```

Comment:
Why make this throw? If content and sender can both be `nil`, why not use `try?`, which would allow you to not throw?